nodeup/pkg/model/protokube.go

```diff
@@ -219,11 +220,23 @@ func (t *ProtokubeBuilder) ProtokubeFlags(k8sVersion semver.Version) *ProtokubeF
 	f := &ProtokubeFlags{
 		Channels:      t.NodeupConfig.Channels,
 		Containerized: fi.Bool(true),
-		EtcdImage:     s(fmt.Sprintf("gcr.io/google_containers/etcd:%s", imageVersion)),
 		LogLevel:      fi.Int32(4),
 		Master:        b(t.IsMaster),
 	}
 
+	// TODO this is dupicate code with etcd model
+	image := fmt.Sprintf("gcr.io/google_containers/etcd:%s", imageVersion)
```

Comment:
Could we make this configurable to allow users to specify either the full address or version within their Cluster Spec? Similar to how it works for `KubernetesVersion`.

Reply:
That is exactly what https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/cluster.go#L176 does. The assests builder uses the container registry setting.

kops will upload the dependencies to you registry during the assests phase and then your cluster can use them.

This work is me wiring everything together.

Reply:
Ah I missed that, ok perfect! 👍